### ISSUE TITLE
The installer posh-windows-386.exe is flagged as a threat.  Request to establish authenticode signature.

### ISSUE THREAD

**Issue by romanrozinov** [JanDeDobbeleer/oh-my-posh#767]
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] I looked for duplicate issues before submitting this one

### Description

As part of the rollout attempt of oh-my-posh to the development team, it is detected as a threat by SentinelOne software.  Vendor's response is that increasing reputation score would help by singing the executable with Microsoft's Autneticode technology, see [SignTool](https://docs.microsoft.com/en-us/windows/win32/seccrypto/signtool). 

Hence the request for enhancement to establish the signature for the installation executable, as opposed to relying on static hashes every release.

### Environment

- Oh my Posh version:  v3.159.0
- Theme: N/A
- Operating System:  Windows 10
- Shell:  Poweshell
- Terminal:  Windows Terminal
  
#### Optional

- posh-git version: v1.0.0-beta4
- git version: version 2.31.1.windows.1

### Steps to Reproduce

1. `Install-Module  oh-my-posh -Force` 

**Expected behavior:** [What you expected to happen]

 Installation should be successful.

**Actual behavior:** [What actually happened]

Installation is aborted due to antivirus flagging the executable.

**Comment by romanrozinov** [comment]
I am aware of #708 but this issue is related to psm1 install script itself which i believe should also be signed, if not already, to allow greater adoption in enterprise environments.

**Comment by JanDeDobbeleer** [comment]
We've talked about code signing before but the certificates are way too expensive for a project such as this. So, unless you can add an exclusion in Sentinel (that should be possible though on company level), or we find a sponsor for said certificate, there's not much we can do for the time being.

**Comment by JanDeDobbeleer** [comment]
@romanrozinov missed your second comment. I'm curious, what happens when you use scoop/winget to install the oh-my-posh executable? Because the module is simply a wrapper for that anyways.

**Comment by mcroach** [comment]
Possible solution to this eventually may be https://sigstore.dev/ managed by the Linux Foundation. In its current form it appears to support signing of Linux/Windows executables. Not sure if in its current form it could also be used to sign powershell modules or if that might be a future option.

**Comment by JanDeDobbeleer** [comment]
@mcroach yup, it was hinted before. Waiting on it to come to life. In theory, having the executable signed should be sufficient.

**Comment by fasterinnerlooper** [comment]
This Twitter thread has some suggestions: https://twitter.com/mehedih_/status/1411969294724968449?s=21

**Comment by ShaneYu** [comment]
Not sure if this is related or not, but McAfee is flagging the .exe as a detected threat containing a Trojan virus.
![image](https://user-images.githubusercontent.com/3437388/132203541-ae72995f-9dfa-4567-8009-4f3245d8b3c1.png)

Edit: I can see it's related and it's also affecting all install methods (PS, winget etc); even downloaded the source and tried to build it and McAfee jumps on it. Really big shame as I love using it at home.

**Comment by JanDeDobbeleer** [comment]
We're eagerly awaiting a sponsor or sigstore coming to life :-)

**Comment by darkvertex** [comment]
@JanDeDobbeleer Did you know there is a handy Github Action for autoscanning releases for viruses via the awesome VirusTotal API?

Maybe it'd be worth adding it to the release workflows eventually:
https://github.com/marketplace/actions/virustotal-github-action

Won't fix your ocasional false positive but it may help you
a) spot which AV engines are flagging your executables, and
b) may help give people more confidence the threat is false if they can click through to a linked virus scan report that shows there is 1 lonely false positive out of 92 antivirus engines utilized.

If you've never heard of VirusTotal before, it's a great tool that scans a file with a ton of commercial and free antiviruses all at once. Here's an [example scan from the 5.16.5 amd64 exe.](https://www.virustotal.com/gui/url/d1a83364c29d251e94cf035ff191f7f78047b1eaf05814d933fbf13b00b3ec53)

**Comment by jonaskuske** [comment]
> We've talked about code signing before but the certificates are way too expensive for a project such as this.

Not sure if that's helpful here, but I've added this affordable OSS certificate to my bookmarks some time ago: https://shop.certum.eu/data-safety/code-signing-certificates/open-source-code-signing-on-simplysign.html

Some limitations apply, don't know if they're a dealbreaker:

[![Screenshot_20220308-025237~2.png](https://user-images.githubusercontent.com/30421456/157150392-8aaf6753-336f-40d2-aa59-81cf8de81645.png)](https://www.support.certum.eu/en/code-signing-required-documents/)

**Comment by MattBDev** [comment]
Since signing certificates are expensive have you considered packing as a MSIX and publishing in the Microsoft Store? It doesn't completely solve the issue since some Windows machines may have a group policy set that blocks the Store but does sign the app.
My company uses Cortex XDR and OhMyPosh was detected as a threat and blocked.

**Comment by JanDeDobbeleer** [comment]
@MattBDev if you have a good guide on how to set this up, let me know. Always down for this.

**Comment by MattBDev** [comment]
@JanDeDobbeleer There is a good guide [here](https://docs.microsoft.com/en-us/windows/msix/desktop/cicd-overview) on how to do it on CI/CD Pipelines. There is also [this link](https://docs.microsoft.com/en-us/windows/msix/package/manual-packaging-root) for doing it over the command line.

**Comment by JanDeDobbeleer** [comment]
I'm actively working on this. Expect this to land within the next week(s) if all goes according to plan.

**Comment by floh96** [comment]
@JanDeDobbeleer btw yesterday i discovered the digicert MVP Program https://www.digicert.com/friends/msmvp/ just fyi

**Comment by JanDeDobbeleer** [comment]
@floh96 I'm actively working on that, pending identity validation but it takes time (planned for next week Friday).

**Comment by mekwall** [comment]
![image](https://user-images.githubusercontent.com/37091/177314491-1f9267eb-8e91-48a8-bb3d-a742b4ebb962.png)

Just a heads up that the latest update is getting flagged by Windows Security.

**Comment by JanDeDobbeleer** [comment]
Keep me in the loop as to how this evolves:

<img width="909" alt="image" src="https://user-images.githubusercontent.com/2492783/177478954-94305343-92a4-4e7f-bcb2-d28ca157793d.png">

**Comment by OranguTech** [comment]
@JanDeDobbeleer - FWIW, it worked for me just fine on a new-ish Win10 install:

```
PS C:\Windows\System32>winget install JanDeDobbeleer.OhMyPosh -s winget
Found Oh My Posh [JanDeDobbeleer.OhMyPosh] Version 8.13.1
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v8.13.1/install-amd64.exe
  ██████████████████████████████  6.55 MB / 6.55 MB
Successfully verified installer hash
Starting package install...
Successfully installed
```

**Comment by mekwall** [comment]
![image](https://user-images.githubusercontent.com/37091/177932208-6dfd1b48-ad19-47fb-a00f-569150517f26.png)

Works fine here as well! No false positives by Windows Security 🎉
